c/main.c

```diff
@@ -70,6 +70,10 @@ enum cpu_feature {
 extern enum cpu_feature g_cpu_features;
 enum cpu_feature get_cpu_features();
 
+#ifdef __TRUSTINSOFT_ANALYZER__
+extern FILE *fp_output;
+#endif
+
```

Comment:
I've been thinking about simplifying this `main.c` file for a while. A lot of different tests flow through here, and as a result it does some weird stuff like buffering its entire input in memory and writing to the private global `g_cpu_features` flags. Originally I wrote it to be a usable by a human on the command line, but at some point I gave up on that. Eventually I'd like to take it apart, but with this change I'd be worried about breaking our TIS tests and not knowing how to fix them.

Could it make sense to just copy the important parts into `tis-ci/test.c`? If the TIS tests need private access to e.g. `g_cpu_features`, I wonder if we could add preprocessor flags or maybe new API functions to support them.

Reply:
Hmm. Basically: do as you please, as you surely know much better than me how this code should be organized :)

To give you some context: my aim here was to deploy TIS-CI on the project in a simple and efficient way, piggy-backing on existing test-suite as much as possible. So frankly I did not even look too much into what was exactly going on in `main.c` and why. I've used it pretty much as-is, because it is the entry point for these test vector tests from `c/test.py`. The only problem that I've encountered was that TIS-CI couldn't compare what was printed on stdout with the expected output read from a file. Hence the only change that I needed to implement was adding the printing to an external file in addition to printing to stdout.

So, answering your question: yes, as long as there exists a function that can be called from `main_wrapper` that outputs the same thing as current `main` in `c/main.c` (to an external file and not only stdout) and which returns the same values, then TIS-CI will be fine. You can move or copy or trim this code as you see fit, e.g. putting it directly in `tis-ci/test.c` or in a new file `tis-ci/main.c`, whichever seems easier to maintain (some minor changes to `tis.config` might be necessary you move the files around, but I can help you with that).